.github/workflows/release.yaml

```diff
@@ -0,0 +1,33 @@
+# SPDX-License-Identifier: MIT
+
+name: release
+
+on:
+  push:
+    # run only against tags
+    tags:
+      - 'v*'
+
+permissions:
+  contents: write
```

Comment:
Are we sure this will work? We know we have a longstanding issue with not having write permission on the repositories as configured by the CNCF.

Reply:
yes it works as in this case the release got pushed to the auger release section. the PR description contains two links where you can see how it was done on my fork.

Reply:
I may not have explained myself correctly. We don't have write permission to the repositories through GitHub actions; we have attempted this several times in different repositories (etcd, bbolt, the website, etc.). And other projects in the CNCF umbrella are facing the same issue :sweat_smile:

Reply:
Good spotting @ivanvc - Yeah `contents: write` is currently not possible due to the k8s org GitHub enterprise policies, ref https://github.com/kubernetes/org/issues/5126#issuecomment-2350687894/

There is a lot of noise about this currently so I will leave this open for a week or two to see if there is a resolution. The backup plan is to turn to a prow based setup for automating releases but that will be much more work.

Thanks @bavarianbidi for putting forward this proposal and for bearing with us while we try to find a way forward.

Reply:
The problem that I found is that even if we set it as a prow job, we don't have access to the trusted cluster that has the secrets with the GitHub tokens. Refer to https://github.com/kubernetes/test-infra/pull/33365#issuecomment-2319069142.

So, I've always seen any automation as a dead end.

I agree with keeping this open to see if we have a resolution soon(ish).

Reply:
that's super sad to hear. But it sounds like a very generic problem to every project under the CNCF enterprise in github. And yes, i agree with you @ivanvc , it's a dead end for release automation.

Reply:
turns out, that it should already work with the `content: write` permission:

xref: https://github.com/kubernetes/org/issues/5126#issuecomment-2431095982

commit in `kubebuilder` repo by adding the permissions into the goreleaser based build process: https://github.com/kubernetes-sigs/kubebuilder/commit/eae8b2182a65df49eb04ed63d6c45043b454905c

Reply:
Thanks for the update @bavarianbidi - The linked updates & slack thread sound promising, let's try it.